### PR TITLE
upstream v5.55.0

### LINKED ISSUE #4093: Upgrade terraform-provider-aws to v5.55.0
Release details: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.55.0

### PR DESCRIPTION
**Pull request by flostadler**
- **Moving ./upstream to v5.55.0**
- **Update patches**
- **./scripts/tidy-all.sh**
- **Make scripts/patch_computed_only.sh honor whitespaces**
- **./script/patch_computed_only.sh**
- **Add mod mappings for new resources**
- **Regenerate schema**
- **Regenerate SDK**

closes #4093

### CODE REVIEW

File: patches/0036-Disable-retry-for-KMS-access-denied-in-lambda.patch [hunk 1]
```diff
@@ -43,71 +42,42 @@ index 0000000000..4625f9e31c
...
-+			o.BaseEndpoint = aws_sdkv2.String(endpoint)
 +
-+			if o.EndpointOptions.UseFIPSEndpoint == aws_sdkv2.FIPSEndpointStateEnabled {
-+				tflog.Debug(ctx, "endpoint set, ignoring UseFIPSEndpoint setting")
-+				o.EndpointOptions.UseFIPSEndpoint = aws_sdkv2.FIPSEndpointStateDisabled
-+			}
-+		}
-+		o.Retryer = conns.AddIsErrorRetryables(cfg.Retryer().(aws_sdkv2.RetryerV2), retry)
-+	}), nil
++	return lambda_sdkv2.NewFromConfig(cfg,
```
Comment (flostadler) [inline patches/0036-Disable-retry-for-KMS-access-denied-in-lambda.patch:46]: There's multiple similar changes to patches in this PR.
Upstream introduces some helpers for configuring the endpoint: https://github.com/hashicorp/terraform-provider-aws/pull/38057/files

File: scripts/patch_computed_only.sh [hunk 2]
```diff
@@ -29,6 +29,6 @@ FILES=$(cd upstream && git grep TagsAttributeComputedOnly |
 
 
 for f in $FILES; do
-    replace="s/names.AttrTagsAll: tftags.TagsAttributeComputedOnly/names.AttrTagsAll: tftags.TagsAttribute/g"
+    replace="s/names.AttrTagsAll:\s+tftags.TagsAttributeComputedOnly/names.AttrTagsAll: tftags.TagsAttribute/g"
```
Comment (flostadler) [inline scripts/patch_computed_only.sh:32]: This now supports different amounts of whitespaces

File: patches/0018-Change-default-descriptions-to-Managed-by-Pulumi.patch [hunk 3]
```diff
@@ -18,17 +18,17 @@ index abc31f0d40..ebf1de5fb3 100644
...
+index fbdca7da91..9cdc01f8ec 100644
 --- a/internal/service/appsync/api_key.go
 +++ b/internal/service/appsync/api_key.go
-@@ -36,7 +36,7 @@ func ResourceAPIKey() *schema.Resource {
+@@ -49,7 +49,7 @@ func resourceAPIKey() *schema.Resource {
  			names.AttrDescription: {
  				Type:     schema.TypeString,
  				Optional: true,
 -				Default:  "Managed by Terraform",
 +				Default:  "Managed by Pulumi",
```
Comment (t0yv0) [inline patches/0018-Change-default-descriptions-to-Managed-by-Pulumi.patch:29]: A bit tedious but we could move this from a patch some Go code running on provider init. Fewer patches, less busywork.

File: patches/0036-Disable-retry-for-KMS-access-denied-in-lambda.patch [hunk 4]
```diff
@@ -43,71 +42,42 @@ index 0000000000..4625f9e31c
...
-diff --git a/names/data/names_data.csv b/names/data/names_data.csv
-index e36b7a9a0c..ebe0f7ed17 100644
---- a/names/data/names_data.csv
-+++ b/names/data/names_data.csv
-@@ -218,7 +218,7 @@ kinesis-video-media,kinesisvideomedia,kinesisvideomedia,kinesisvideomedia,,kines
- kinesis-video-signaling,kinesisvideosignaling,kinesisvideosignalingchannels,kinesisvideosignaling,,kinesisvideosignaling,,kinesisvideosignalingchannels,KinesisVideoSignaling,KinesisVideoSignalingChannels,,1,,,aws_kinesisvideosignaling_,,kinesisvideosignaling_,Kinesis Video Signaling,Amazon,,x,,,,,Kinesis Video Signaling,,,,
- kms,kms,kms,kms,,kms,,,KMS,KMS,,,2,,aws_kms_,,kms_,KMS (Key Management),AWS,,,,,,,KMS,ListKeys,,,
- lakeformation,lakeformation,lakeformation,lakeformation,,lakeformation,,,LakeFormation,LakeFormation,,,2,,aws_lakeformation_,,lakeformation_,Lake Formation,AWS,,,,,,,LakeFormation,ListResources,,,
--lambda,lambda,lambda,lambda,,lambda,,,Lambda,Lambda,,,2,,aws_lambda_,,lambda_,Lambda,AWS,,,,,,,Lambda,ListFunctions,,,
-+lambda,lambda,lambda,lambda,,lambda,,,Lambda,Lambda,x,,2,,aws_lambda_,,lambda_,Lambda,AWS,,,,,,,Lambda,ListFunctions,,,
```
Comment (t0yv0) [inline patches/0036-Disable-retry-for-KMS-access-denied-in-lambda.patch:110]: Wish GitHub knew how to display CSV  diffs properly.. This is terrible.

**Review by t0yv0 — APPROVED** [review]
LGTM

File: patches/0018-Change-default-descriptions-to-Managed-by-Pulumi.patch (hunk 3 above)
Comment (flostadler) [reply patches/0018-Change-default-descriptions-to-Managed-by-Pulumi.patch:29]: Do we have a ticket for this already?

**Pull request merged**